src/components/ImagesCarousel.js

```diff
@@ -1,24 +1,31 @@
 import PropTypes from 'prop-types';
-import Carousel from 'react-native-snap-carousel';
-import React, { useCallback, useContext } from 'react';
-import { ActivityIndicator, StyleSheet } from 'react-native';
+import React, { useCallback, useContext, useState } from 'react';
 import { Query } from 'react-apollo';
+import { ActivityIndicator, StyleSheet, TouchableOpacity, View } from 'react-native';
+import Carousel from 'react-native-snap-carousel';
 
-import { colors } from '../config';
+import { colors, Icon, normalize } from '../config';
 import { graphqlFetchPolicy, imageWidth, isActive, shareMessage } from '../helpers';
-import { getQuery } from '../queries';
-import { OrientationContext } from '../OrientationProvider';
-import { NetworkContext } from '../NetworkProvider';
 import { useRefreshTime } from '../hooks';
+import { NetworkContext } from '../NetworkProvider';
+import { OrientationContext } from '../OrientationProvider';
+import { getQuery } from '../queries';
+import { SettingsContext } from '../SettingsProvider';
 
 import { ImagesCarouselItem } from './ImagesCarouselItem';
 import { LoadingContainer } from './LoadingContainer';
 
 export const ImagesCarousel = ({ data, navigation, refreshTimeKey, aspectRatio }) => {
   const { dimensions } = useContext(OrientationContext);
   const { isConnected, isMainserverUp } = useContext(NetworkContext);
-
+  const { globalSettings } = useContext(SettingsContext);
+  const { settings = {} } = globalSettings;
+  const { sliderPauseButton } = settings;
+  const { show: showSliderPauseButton = false, size: sizeSliderPauseButton = 25 } =
+    sliderPauseButton || {};
```

Comment:
i guess `const { sliderPauseButton = {} } = settings;` would replace `... =
    sliderPauseButton || {};` in a more readable way

Reply:
https://github.com/smart-village-solutions/smart-village-app-app/commit/6117184332afa80c19596a82cb1e0cab2aa57624